packages/medusa/src/utils/transformers/tree.ts

```diff
@@ -0,0 +1,25 @@
+import { pick } from "lodash"
+
+// TODO: When we implement custom queries for tree paths in medusa, remove the transformer
+// Adding this here since typeorm tree repo doesn't allow configs to be passed
+// onto its children nodes. As an alternative, we are transforming the data post query.
+export function transformTreeNodesWithConfig(object, config) {
+  const selects = (config.select || []) as string[]
+  const relations = (config.relations || []) as string[]
+  const selectsAndRelations = selects.concat(relations)
+
+  if (object.parent_category) {
```

Comment:
todo: just a reminder for additional guard

Reply:
done, take a look :)